docs/Advanced-Topics-Custom-Block-Render.md

```diff
@@ -0,0 +1,158 @@
+---
+id: advanced-topics-custom-block-render-map
+title: Custom Block Rendering
+layout: docs
+category: Advanced Topics
+next: advanced-topics-block-components
+permalink: docs/advanced-topics-custom-block-render-map.html
+---
+
+This article discusses how to customize Draft default block rendering.
+The block rendering is used to define supported block types and their respective
+renderers, as well as converting pasted content to known Draft block types.
+
+When pasting content or when using the
+[convertFromHTML](https://facebook.github.io/draft-js/docs/api-reference-data-conversion.html#convertfromhtml)
+Draft will then convert the pasted content to the respective block rendering type
+by matching the Draft block render map with the matched tag.
+
+## Draft default block render map
+
+|  HTML element   |            Draft block type             |
+| --------------- | --------------------------------------- |
+|     `<h1/>`     |               header-one                |
+|     `<h2/>`     |               header-two                |
+|     `<h3/>`     |              header-three               |
+|     `<h4/>`     |               header-four               |
+|     `<h5/>`     |               header-five               |
+|     `<h6/>`     |               header-six                |
+|     `<h6/>`     |               header-six                |
+| `<blockquote/>` |               blockquote                |
+|    `<pre/>`     |               code-block                |
+|   `<figure/>`   |                 atomic                  |
+|     `<li/>`     | unordered-list-item,ordered-list-item** |
+|    `<div/>`     |               unstyled***               |
+
+\*\* - Block type will be based on the parent `<ul/>` or `<ol/>`
+
+\*\*\* -  Any block that is not recognized by the block redering mapping will be treated as unstyled
+
+## Configuring block render map
+
+Draft default block render map can be overwritten, by passing an
+[Immutable Map](http://facebook.github.io/immutable-js/docs/#/Map) to
+the editor blockRender props.
+
+*example of overwritting default block render map:*
+
+```js
+// The example below deliberatly only allows
+// 'heading-two' and 'paragraph' as valid block types and
+// updated the unstyled element to become a paragraph.
+const blockRenderMap = Immutable.Map({
+  'header-two': {
+   element: 'h2'
+  },
+  'paragraph': {
+   element: 'p'
+  },
+  'unstyled': {
+   element: 'p'
```

Comment:
This was a feature implemented prior to react 0.15 when this warning did not exist. will update the docs to show a different example